pkg/cmd/server/api/types.go

```diff
@@ -450,6 +450,11 @@ type MasterNetworkConfig struct {
 	// CIDR will be rejected. Rejections will be applied first, then the IP checked against one of the allowed CIDRs. You
 	// should ensure this range does not overlap with your nodes, pods, or service CIDRs for security reasons.
 	ExternalIPNetworkCIDRs []string
+	// IngressIPNetworkCIDRs controls the range to assign ingress ips from for services of type LoadBalancer on bare
+	// metal. If empty, ingress ips will not be assigned. It may contain a single CIDR that will be allocated from.
+	// For security reasons, you should ensure that this range does not overlap with the CIDRs reserved for external ips,
+	// nodes, pods, or services.
+	IngressIPNetworkCIDRs []string
```

Comment:
I don't think we need full flexibility yet.  Adding a new field is always
possible if necessary (and doing conversion).

On Mon, Aug 1, 2016 at 4:30 PM, Maru Newby <notifications@github.com> wrote:

> In pkg/cmd/server/api/types.go
> <https://github.com/openshift/origin/pull/9454#discussion_r73047346>:
>
> > @@ -450,6 +450,11 @@ type MasterNetworkConfig struct {
> >  	// CIDR will be rejected. Rejections will be applied first, then the IP checked against one of the allowed CIDRs. You
> >  	// should ensure this range does not overlap with your nodes, pods, or service CIDRs for security reasons.
> >  	ExternalIPNetworkCIDRs []string
> > +	// IngressIPNetworkCIDRs controls the range to assign ingress ips from for services of type LoadBalancer on bare
> > +	// metal. If empty, ingress ips will not be assigned. It may contain a single CIDR that will be allocated from.
> > +	// For security reasons, you should ensure that this range does not overlap with the CIDRs reserved for external ips,
> > +	// nodes, pods, or services.
> > +	IngressIPNetworkCIDRs []string
>
> I assumed it would be desirable to be able to specify the range more
> flexibly in the future (in the same way as ExternalIPNetworkCIDRs), and
> that it would be preferable to relax the restriction on a single CIDR
> rather than having to change the name and type of the parameter to do so.
> Is future flexibility desirable or should I do as you suggest?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/origin/pull/9454/files/d9ba22a6fc7d91c617d52a922851c4c6acdbb65e#r73047346>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABG_pwQF_EH2D__8RzWfoQSSjSRqdWlSks5qbld6gaJpZM4I6YVl>
> .
>